biopolymers.csl

```diff
@@ -0,0 +1,253 @@
+      <choose>
+        <if type="article-journal review" match="any">
+          <group delimiter=" ">
+            <text variable="container-title" form="short" strip-periods="true"/>
+            <group delimiter=", ">
+              <text macro="issued"/>
+              <choose>
+                <if variable="volume">
+                  <group delimiter=" ">
+                    <text variable="volume"/>
+                  </group>
+                </if>
+                <else-if variable="issue">
+                  <group delimiter=" ">
+                    <text term="issue" form="short" text-case="capitalize-first"/>
+                    <text variable="issue"/>
+                  </group>
+                </else-if>
+              </choose>
+              <text variable="page"/>
+            </group>
+          </group>
+        </if>
+        <else-if type="article-magazine article-newspaper article" match="any">
+          <group delimiter=" ">
+            <text variable="container-title" suffix="."/>
+            <text macro="edition"/>
+            <text macro="publisher"/>
+            <group delimiter=", ">
+              <text macro="full-issued"/>
+              <text macro="pages"/>
+            </group>
+          </group>
+        </else-if>
+        <else-if type="thesis">
+          <group delimiter=", ">
+            <group delimiter=". ">
+              <text variable="title"/>
+              <text variable="genre"/>
+            </group>
+            <text macro="publisher"/>
+            <text macro="issued"/>
+            <text macro="volume"/>
+            <text macro="pages"/>
+          </group>
+        </else-if>
+        <else-if type="bill book graphic legal_case legislation motion_picture report song" match="any">
+          <group delimiter="; ">
+            <group delimiter=", ">
+              <text variable="title"/>
+              <text macro="edition"/>
+            </group>
+            <text macro="series" prefix=" "/>
+            <choose>
+              <if type="report">
+                <group delimiter=" ">
+                  <text variable="genre"/>
+                  <text variable="number"/>
+                </group>
+              </if>
+            </choose>
+            <group delimiter=", ">
+              <text macro="publisher"/>
+              <text macro="issued"/>
+            </group>
+            <group delimiter=", ">
+              <text macro="volume"/>
+              <text macro="pages"/>
+            </group>
+          </group>
+        </else-if>
+        <else-if type="patent">
+          <group delimiter=", ">
+            <group delimiter=". ">
+              <text variable="title"/>
+              <text variable="number"/>
+            </group>
+            <date variable="issued" form="text"/>
+          </group>
+        </else-if>
+        <else-if type="chapter paper-conference entry-dictionary entry-encyclopedia" match="any">
+          <group delimiter="; ">
```

Comment:
this got a bit messy. Replace 211-224 with

```
<group delimiter="; ">
  <text macro="book-container"/>
  <text macro="series"/>
  <group delimiter=", ">
     <text macro="publisher"/>
     <text macro="issued"/>
  </group>
  <text macro="volume"/>
  <text macro="pages"/>
</group>
```

which should do the same, just more cleanly and more robustly.